repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -2191,6 +2211,62 @@ private void addEdgesToContext(String guid, List<Object> newElementsCreated, Lis
         }
     }
 
+    private void addInternalProductAttr(AttributeMutationContext ctx, List<Object> createdElements, List<AtlasEdge> deletedElements, List<Object> currentElements) throws AtlasBaseException {
+        MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("addInternalProductAttrForAppend");
+        AtlasVertex toVertex = ctx.getReferringVertex();
+        String toVertexType = getTypeName(toVertex);
+
```

Comment:
return immediately in case all 3 createdElements, deletedElements, currentElements are empty
otherwise we will end up removing attribute which will cause bug in the flow